### ISSUE TITLE
"jpm sign" returns an XPI which is just a JSON blob: {"detail": "Not found"}

### ISSUE THREAD

**Issue by andymckay** [mozilla-jetpack/jpm#515]
From https://bugzilla.mozilla.org/show_bug.cgi?id=1255503

When I sign my addon from commmand line, I get back an XPI which is just a JSON blob with just "{"detail": "Not found"}" in the file.

```
$ jpm sign --api-key user:<some_user> --api-secret <some_password>
Creating XPI
JPM [info] XPI created at /var/folders/b3/wz8byszx531_gwwzp7bzx8yw0000gn/T/tmp-unsigned-xpi-61428W7uEa2TtEBjc/@mls-offline-0.0.13.xpi (396ms)
Created XPI at /var/folders/b3/wz8byszx531_gwwzp7bzx8yw0000gn/T/tmp-unsigned-xpi-61428W7uEa2TtEBjc/@mls-offline-0.0.13.xpi
JPM [info] Created XPI for signing: /var/folders/b3/wz8byszx531_gwwzp7bzx8yw0000gn/T/tmp-unsigned-xpi-61428W7uEa2TtEBjc/@mls-offline-0.0.13.xpi
Validating add-on [.............................................................................................................................]JPM [info] Validation results: https://addons.mozilla.org/en-US/developers/upload/56e8dcc50384444384406ae3558b91bf
Downloading signed files: 100%
JPM [info] Downloaded:
JPM [info]     ./privacy_enhanced_locations-0.0.13-an+fx.xpi
JPM [info] SUCCESS

$ cat privacy_enhanced_locations-0.0.13-an+fx.xpi
{"detail": "Not found"}
```

This occurs for my addon in https://github.com/crankycoder/libmarisa/tree/dev/libraries/marisa/addon

Building the addon with just "jpm xpi" will give me a normal (but obviously unsigned) XPI:

$ ls -l *.xpi
-rw-r--r--  1 victorng  staff  5737141 10 Mar 12:54 @mls-offline-0.0.13.xpi
-rw-r--r--  1 victorng  staff       23 10 Mar 12:48 privacy_enhanced_locations-0.0.13-an+fx.xpi

**Comment by kumar303** [comment]
@crankycoder can you re-run the command as `jpm sign --verbose ...` and attach the log?

**Comment by kumar303** [comment]
at the very least, jpm probably needs to check for 404 responses on the final download URL

**Comment by crankycoder** [comment]
Here's my verbose log - the beginning got truncated though. [sign.txt](https://github.com/mozilla-jetpack/jpm/files/204994/sign.txt)

**Comment by crankycoder** [comment]
I think this bug is probably related to https://github.com/mozilla/addons/issues/62

**Comment by kumar303** [comment]
Heh, oops, I guess `--verbose` doesn't log response status. Anyway, the sign function should still check for 404 responses and show an error. FYI: we should fix https://github.com/mozilla-jetpack/jpm/issues/519 first.

**Comment by kumar303** [comment]
corresponding `sign-addon` bug: https://github.com/mozilla/sign-addon/issues/74

**Comment by kumar303** [comment]
Fixed in https://github.com/mozilla-jetpack/jpm/commit/89c6be00b9b88c95277bab8c17fffbb36f664d62

This was released in 1.1.2 https://github.com/mozilla-jetpack/jpm/releases/tag/1.1.2
